Showcase-iOS/Login/View/LoginViewController.swift

```diff
@@ -33,6 +34,9 @@ class LoginViewController: UIViewController {
         loginScrollView.keyboardDismissMode = .interactive
         alertController = UIAlertController(title: "Can't log you in", message: "", preferredStyle: .alert)
         alertController?.addAction(retryAction)
+        if userDefaults.bool(forKey: UserDefaultsKeys.isLoggedIn.rawValue) {
+            self.showSuccess()
```

Comment:
This should be in the Presenter, and you should write tests for this please. So protocol this implementation out please.